src/python/module/smartredis/dataset_utils.py

```diff
@@ -0,0 +1,141 @@
+            :type dim_names: list[str]
+            :param coord_names: coordinate field names. Defaults to None.
+            :type coord_names: list[str], optional
+            :param attr_names: attribute field names. Defaults to None.
+            :type attr_names: list[str], optional
+        """        
+        
+        typecheck(dataset, "dataset", Dataset)
+        typecheck(data_names, "data_names", list)
+        for data_name in data_names:
+            typecheck(data_name, "data_name", str)
+        typecheck(dim_names, "dim_names", list)
+        for dim_name in dim_names:
+            typecheck(dim_name, "dim_name", str)
+        if coord_names:
+            typecheck(coord_names, "coord_names", list)
+        if attr_names: 
+            typecheck(attr_names, "attr_names", list)        
+        
+        args = ['dim_names','coord_names','attr_names'] 
+
+        for d in data_names:  
+            dataset.add_meta_string("_xarray_data_names",d)
+            for arg in args:
+                if isinstance(eval(arg),list):
+                    values = []
+                    for val in eval(arg): 
+                        values.append(val)
+                    arg_field=",".join(values)
+                    dataset.add_meta_string(f"_xarray_{d}_{arg}",arg_field)
+                else:
+                    if eval(arg) != None: 
+                        dataset.add_meta_string(f"_xarray_{d}_{arg}",eval(arg))
+                        print(f"_xarray_{d}_{arg}")
+                    else: 
+                        dataset.add_meta_string(f"_xarray_{d}_{arg}","null")
+
+    @staticmethod
+    def transform_to_xarray(dataset): 
+        """Transform a SmartRedis Dataset, with the appropriate metadata,
+        to an Xarray Dataarray 
+
+        :param dataset: a Dataset instance
+        :type dataset: Dataset
+        
+        :return: a dictionary of keys as the data field name and the
+        value as the built Xarray DataArray constructed using 
+        fieldnames and appropriately formatted metadata. 
+        rtype: dict
+        """
+        typecheck(dataset, "dataset", Dataset)
+        
+        coord_dict= {}
+        coord_final = {}
+        variable_names = dataset.get_meta_strings("_xarray_data_names")
+             
+        for j,d in list(permutations(variable_names,2)):
+            for k in dataset.get_meta_strings(f"_xarray_{j}_coord_names")[0].split(","):
+                if d == k:
+                    # Remove coordinate data names from data names - may refactor this to make more clear
+                    variable_names.remove(d)
+                    # Get coordinate dimensions in the appropriate format for Xarray 
+                    coord_dims = []
+                    for coord_dim_field_name in dataset.get_meta_strings(f"_xarray_{d}_dim_names")[0].split(","):
+                        coord_dims.append(dataset.get_meta_strings(coord_dim_field_name)[0])
```

Comment:
Can this be built more cleanly with a lambda expression? (Same applies to coord_attrs)